src/main/java/de/tum/in/www1/artemis/web/rest/programming/ProgrammingExerciseParticipationResource.java

```diff
@@ -285,6 +286,28 @@ public ResponseEntity<List<CommitInfoDTO>> getCommitHistoryForParticipationRepo(
         return ResponseEntity.ok(commitInfo);
     }
 
+    /**
+     * GET /programming-exercise-participations/{participationId}/user : Get the user of a programming exercise participation.
+     * Here we check if the requesting user is at least a teaching assistant for the exercise
+     * If the user is a student, it checks if the user is the owner of the participation.
+     *
+     * @param participationId the id of the participation for which to retrieve the user
+     * @return the ResponseEntity with status 200 (OK) and with body a UserDTO, or status 404 if the participation or user does not exist or is a team participation
+     *
+     */
+    @GetMapping("programming-exercise-participations/{participationId}/user")
+    @EnforceAtLeastStudent
+    public ResponseEntity<UserDTO> getUserForParticipation(@PathVariable long participationId) {
+        ProgrammingExerciseStudentParticipation participation = programmingExerciseStudentParticipationRepository.findByIdElseThrow(participationId);
+        participationAuthCheckService.checkCanAccessParticipationElseThrow(participation);
+
+        if (participation.getTeam().isPresent()) {
+            return ResponseEntity.notFound().build();
+        }
```

Comment:
Could you throw an exception here (e.g. `BadRequestAlertException`) that explains what went wrong with the request? I feel like the error should indicate that no single user can be extracted from a team participation.

Reply:
Also logging the error mesage is a good idea